frontend/src/features/books/EditBookForm.jsx

```diff
@@ -0,0 +1,110 @@
+    const navigate = useNavigate()
+
+    useEffect(() => {
+        const fetchCurrentBook = async () => {
+            try {
+                const response = await fetch(`${API_URL}/${id}`)
+                if (response.ok) {
+                    const json = await response.json()
+                    setBook(json)
+                    setTitle(json.title)
+                    setAuthor(json.author)
+                    setDescription(json.description)
+                } else {
+                    throw response
+                }
+            } catch (e) {
+                console.log("an error occurred")
+            } finally {
+                setLoading(false)
+            }
+        }
+        fetchCurrentBook()
+    }, [id])
+
+    const handleSubmit = async (e) => {
+        e.preventDefault()
+        const updatedBook = {
+            ...book,
+            title,
+            author,
+            description
+        }
+        try {
+            const response = await fetch(`${API_URL}/${id}`, {
+                method: "PUT",
+                headers: {
+                    "Content-Type": "application/json"
+                },
+                body: JSON.stringify(updatedBook)
+            })
+            if (response.ok) {
+                const json = await response.json()
+                console.log("Success: ", json)
+                navigate(`/books/${id}`)
+            } else {
+                throw response
+            }
+        } catch (e) {
+            console.log("an error occurred", e)
+        }
+    }
+
+    if (!book) return <h3>Loading...</h3>
+
+    return (
+        <div>
+            <h3>Edit your book here</h3>
+            <form onSubmit={handleSubmit}>
+                <div>
+                    <label htmlFor='book-title'>Title</label>
+                    <br></br>
+                    <input
+                        type='text'
+                        id='book-title'
+                        value={title}
+                        onChange={(e) => setTitle(e.target.value)}
+                    ></input>
+                </div>
+                <div>
+                    <label htmlFor='book-author'>Author</label>
+                    <br></br>
+                    <input
+                        type='text'
+                        id='book-author'
+                        value={author}
+                        onChange={(e) => setAuthor(e.target.value)}>
+                    </input>
+                </div>
+                <div>
+                    <label htmlFor='book-description'>Description</label>
+                    <br></br>
+                    <input
+                        type='text'
+                        id='book-description'
+                        value={description}
+                        onChange={(e) => setDescription(e.target.value)}>
+                    </input>
+                </div>
+                <div>
+                    <button type='submit'>Save</button>
```

Comment:
@nshelokar-vf, can we have 2 spaces for indentation?

Reply:
Sure sir.